### ISSUE TITLE
issue with cacheManager

### ISSUE THREAD

**Issue by vincent3569** [zenphoto/zenphoto#1180]
Hi

after some tests, I think there is an issue with cacheManager/cacheImages.

step to reproduce:
- delete cache on you server
- use basic theme, with standard options on thumbnails (cropped to 100*100)
- use cacheManager on an album and generate only thumb size of the theme (basic for example)
- have a look on the generated pictures in cache (for example, I have this one: img_8991_100_w100_h100_thumb.jpg and thumbs are uncropped
- in front end, the thumb seems to be squared, but this is only by css rules
![img_8991_100_w100_h100_thumb](https://user-images.githubusercontent.com/1224809/43036450-9b237516-8d02-11e8-9c99-307a4d73af45.jpg)

now:
- delete cache again
- visit you albums from the front end
- cache images are generated on the fly, and then they are correct.
- with my exemple, I have this one: img_8991_100_w100_h100_thumb.jpg and thumbs are cropped

![img_8991_100_w100_h100_thumb](https://user-images.githubusercontent.com/1224809/43036499-66c633f2-8d03-11e8-91b8-2163a3ca80bf.jpg)

**Comment by acrylian** [comment]
You did use the support build? Just asking becasue I saw that earlier as far as I can remember. In my attempt with basic - same values - I got two different thumbs by cachemanager and the default one.

**Comment by vincent3569** [comment]
yes of course, I use support build.
issue maybe related with this commit https://github.com/zenphoto/zenphoto/commit/cef851e783890e9f9368f636ec9326d69eeed92b ?

**Comment by acrylian** [comment]
Everything is possible, but I did that as it then fixed things. But all is possible, I will review as soon as I have time.

**Comment by fretzl** [comment]
@vincent3569, what exactlty are the cacheManager settings in your example?
I think it looks like this:
![cachemanager-basic](https://user-images.githubusercontent.com/2480231/43306706-4fdfdb2a-917c-11e8-87e8-aa77c8a37778.png)

**Comment by vincent3569** [comment]
Yes, you are right, this is the cacheManager setting for basic thumbnails

![capture](https://user-images.githubusercontent.com/1224809/43307020-5ae3e182-917d-11e8-93ee-f8d154e44bef.PNG).

with theses settings, cacheManager gives uncropped thumbs like this one
![img_8991_100_w100_h100_thumb](https://user-images.githubusercontent.com/1224809/43308040-6f6677ac-9180-11e8-8377-2c0b579b4f55.jpg)

**Comment by acrylian** [comment]
I will review this as I actually modeled it after the image class thumb function. Probably the default sizes should not be stored as cachemanager sizes and this should internally be just a on/off option instead so the actual option values are used directly.

To get a 100x100 cropped thumb the cropwidth/cropheight parameter also needs to be set I think. The sizes is not needed actually. As said getting the right sizes is a bit 

Our site's theme uses this for the thumbs:
`printCustomSizedImage(getBareImageTitle(), NULL, 255, 128, 255, 128, NULL, NULL, "thumbnail", NULL, true, false); `
So no size parameter set at all.

**Comment by vincent3569** [comment]
as describe above, if I clear the cached images, and visit my album with basic theme, cache images are created on the fly and thumbs are cropped and correct (with the same name img_8991_100_w100_h100_thumb for the example)
![img_8991_100_w100_h100_thumb](https://user-images.githubusercontent.com/1224809/43036499-66c633f2-8d03-11e8-91b8-2163a3ca80bf.jpg)

basic uses this code for image thumbs in album.php `<?php printImageThumb(getAnnotatedImageTitle()); ?>` and for my own theme, I use the same function.

**Comment by acrylian** [comment]
Yeah, as said there will be a bug in the default handling, probably related to not using the option values directly but the values stored in the cachemanager custom sizes. I will review it but it is not always the most convenient code to review ;-)
